crates/snapbox/src/data.rs

```diff
@@ -145,6 +135,43 @@ impl Data {
             DataInner::Text(data) => data.clone().into_bytes(),
         }
     }
+
+    pub fn coerce(self, format: DataFormat) -> Result<Self, crate::Error> {
+        match format {
+            DataFormat::Binary => Ok(Self::binary(self.to_bytes())),
+            DataFormat::Text => match self.try_string() {
+                Some(data) => Ok(Self::text(data)),
+                None => Err(crate::Error::new(format!(
+                    "could not coerce `{}` into `Text`",
+                    self.format()
+                ))),
+            },
+        }
+    }
+
+    pub fn try_string(&self) -> Option<String> {
```

Comment:
What are the expected semantics of this?

Reply:
This should output a `String` if it can. It takes its logic from `try_text`

Reply:
For example, what will this do with structured data?  Is this outputting text or is this outputting a string?